src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/Persistable.java

```diff
@@ -67,6 +68,22 @@
   <T> @Nullable T read(Class<T> clz)
       throws NoSuchMethodException, IllegalAccessException, InvocationTargetException, InstantiationException, DataAccessException;
 
+
+  /**
+   * This API reads all the rows from the table corresponding to the maximum value in the field Object.
+   * @param clz The Class whose Object is desired.
+   * @param <T> The generic type of the class.
+   * @param field DSL field for which the maximum value is desired.
+   * @return A List of instantiated Objects of the class with the fields populated with the data from the corresponding rows in the table.
+   * @throws NoSuchMethodException If the expected setter does not exist.
+   * @throws IllegalAccessException If the setter is not Public
+   * @throws InvocationTargetException If invoking the setter by reflection threw an exception.
+   * @throws InstantiationException Creating an Object of the class failed for some reason.
+   * @throws DataAccessException Thrown by the DB layer.
+   */
+  <T, E> @Nullable List<T> readAllForMaxField(Class<T> clz, Field<E> field)
```

Comment:
This method is called from the RestHandler class. For that class to construct a Field and send it over, couples the request handler with the schema on the persistence layer, moreover with jooq, the library. Can we have them specify the name of the field and we convert the name to Field in the implementer of this method ?

Reply:
I have separated the Handler and the persistence layer and removed the dependency of Jooq from the Request Handler.